### ISSUE TITLE
Experimental keys on scalar is now forbidden

### ISSUE THREAD

**Issue by eserte** [rizen/SVG-Estimate#1]
Tests fail with perl 5.24.0 and later (and also with 5.12 and earlier):

```
Smartmatch is experimental at lib/SVG/Estimate.pm line 261.
Smartmatch is experimental at lib/SVG/Estimate.pm line 264.

#   Failed test 'use SVG::Estimate;'
#   at t/50_estimate.t line 6.
#     Tried to use 'SVG::Estimate'.
#     Error:  Experimental keys on scalar is now forbidden at lib/SVG/Estimate.pm line 296.
# Compilation failed in require at t/50_estimate.t line 6.
# BEGIN failed--compilation aborted at t/50_estimate.t line 6.
Can't locate object method "file_path" via package "SVG::Estimate" at lib/SVG/Estimate.pm line 218.
# Tests were run but no plan was declared and done_testing() was not seen.
# Looks like your test exited with 255 just after 2.
t/50_estimate.t ........................... 
Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 1/2 subtests 
```

Additionally, there are smartmatch warnings which probably should be ceased.

**Comment by rizen** [comment]
Smartmatch is awesome! Why do you hate smartmatch?

**Comment by rizen** [comment]
fixed in 1.0003

**Comment by eserte** [comment]
I think you fixed only the smartmatch warnings. The fatal error "Experimental keys on scalar is now forbidden" is still in 1.0003 (perl >= 5.24, similar error with perl <= 5.12).

**Comment by rizen** [comment]
I have no idea what that means. I don't use Perl 5.24.

**Comment by eserte** [comment]
It means that this is now illegal:

```
my $hash = {foo => "bar"};
my @keys = keys $hash;
```

You have to write instead:

```
my @keys = keys %$hash;
```

(like this was the case with perl 5.12 and earlier)

**Comment by perlDreamer** [comment]
JT, I patched that and committed it, but didn't bump the version or do a release.

**Comment by rizen** [comment]
fixed in 1.0004
